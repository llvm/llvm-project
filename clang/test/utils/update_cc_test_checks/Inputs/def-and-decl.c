// Check that the CHECK lines are generated before the definition and not the declaration
// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu %s -emit-llvm -o - | FileCheck %s

int foo();

void empty_function();

int main() {
  empty_function();
  return foo();
}

int foo() {
  return 1;
}

void empty_function() {}
