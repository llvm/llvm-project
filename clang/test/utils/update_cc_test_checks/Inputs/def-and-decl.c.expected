// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py
// Check that the CHECK lines are generated before the definition and not the declaration
// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu %s -emit-llvm -o - | FileCheck %s

int foo();

void empty_function();

// CHECK-LABEL: @main(
// CHECK-NEXT:  entry:
// CHECK-NEXT:    [[RETVAL:%.*]] = alloca i32, align 4
// CHECK-NEXT:    store i32 0, i32* [[RETVAL]], align 4
// CHECK-NEXT:    call void @empty_function()
// CHECK-NEXT:    [[CALL:%.*]] = call i32 @foo()
// CHECK-NEXT:    ret i32 [[CALL]]
//
int main() {
  empty_function();
  return foo();
}

// CHECK-LABEL: @foo(
// CHECK-NEXT:  entry:
// CHECK-NEXT:    ret i32 1
//
int foo() {
  return 1;
}

// CHECK-LABEL: @empty_function(
// CHECK-NEXT:  entry:
// CHECK-NEXT:    ret void
//
void empty_function() {}
